5 period selection

RoddyMacrae opened the pull request:
# Summary
RDRP-5 deliverables:

- Create a user config file - in TOML
- Put comments in TOML to inform users how to adjust
- Write the code to import it as variable `start_period`, `end_period`
- Convert to type 'date'
- Update working document for users

Below are the steps done to resolve the ticket:

- Added comments to the userconfig.toml file and tried to be descriptive.
- Created a new function called 'period_select()' in the main.py. This returns two datetime.date objects: the start_period and end_period as defined in the userconfig.toml. 
- Documentation was updated with a screenshot of the current userconfig.toml file.

Realised that toml.load returns a dictionary object. The keys are the '[<title>]' objects in the toml file, and the values themselves are dictionaries containing the relevant key-value info under the '[<title>]' lines in userconfig.toml. These sub-dictionaries can then be accessed to get relevant info, such as start_period or end_period.

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes #


#### Code

- [x] **Code runs** The code runs on my machine and/or CDSW
- [x] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [x] **Requirements** My/our code functions according to the requirements of the ticket
- [x] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [x] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script) 
- [x] **Clean Code**
    - [x] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [x] Code passess flake8 linting check
    - [x] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [x] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)


#### Documentation

Any new code includes all the following forms of documentation:

- [x] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [x] Includes `Args` and `returns` for all major functions 
    - [x] The docstring details data types
- [x] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [x] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

jwestw commented on src/main.py:
```diff
@@ -1,7 +1,16 @@
+    return toml.load("/home/cdsw/research-and-development/config/userconfig.toml")
+
+def period_select():
+    """Selects period defined by user in userconfig.toml"""
```
Follow the [style guide](https://google.github.io/styleguide/pyguide.html) on creating docstrings. :+1:

jwestw reviewed (changes requested):
The path to the toml file did not work for me again. I corrected it (in a way that worked for me) and the functions you'd created worked. 

I checked the types returned and they are `datetime.date` which seems correct. 

I think this is the wrong place for these functions to live. Shall we discuss as a team where best to place them? 

Apart from the docstrings, this seems good.

RoddyMacrae commented:
I have corrected the path of the TOML file (userconfig.toml) to "/home/cdsw/research-and-development/config/userconfig.toml" so it should work and run correctly now. 

I have also added a fully descriptive docstring to both the toml_parser() and the period_select() functions. I believe this should cover all previous concerns.

jwestw commented on src/main.py:
```diff
@@ -1,8 +1,40 @@
+    
+        (datetime.date(1990, 10, 10), datetime.date(2000, 10, 5))
+    """    
+    period_dict = toml_parser()["period"]
+    return period_dict["start_period"], period_dict["end_period"]
```
It's not a strict rule, but I suggest leaving a blank line between your process code and the return statement.

RoddyMacrae replied:
Latest commit creates a blank line between each return statement and its corresponding process code.

jwestw reviewed (changes requested):
Please also resolve any conflicts

RoddyMacrae commented on src/main.py:
```diff
@@ -1,8 +1,40 @@
+        {'hive_db': 'hive.db', 'tableName': 'name.table'}, 'outlier_correction': 
+        {'location': 'D:/', 'fileName': 'outliers.txt', 'bool': True}}
+    """    
+    return toml.load("/home/cdsw/research-and-development/config/userconfig.toml")
```
This has been resolved.

jwestw reviewed (approved):
Approving.
